### PR TITLE
repos: Introduce repo.stars column for Github and Gitlab repos

### PR DESCRIPTION
**Pull request by tsenart**
This change set is a precursor to efficiently sorting repos by star count, which is our first iteration of search results ranking. 

Denormalizing this data from the `metadata` column allows us to maintain consistency in the `ReposStore.List` method, which accepts an `OrderBy` option that expects a simple column field. The alternative of casing the repo type and the JSON access path at read time introduces too much additional complexity and is finicky to index properly.

I have tested the most simple forms of the repo listing queries we will do (in a clone of the prod database), and the results are great:

Screen Shot 2021-06-03 at 20 13 06

Follow up PRs will make use of this field for ranking `repo` type search results and suggestions.

Part of #21642

### CODE REVIEW

File: internal/extsvc/gitlab/projects.go [hunk 1]
```diff
@@ -28,6 +28,8 @@ type Project struct {
 	Visibility        Visibility     `json:"visibility"`                    // "private", "internal", or "public"
 	ForkedFromProject *ProjectCommon `json:"forked_from_project,omitempty"` // If non-nil, the project from which this project was forked
 	Archived          bool           `json:"archived"`
+	StarCount         int            `json:"star_count"`
+	ForksCount        int            `json:"forks_count"`
```
Comment (tsenart) [inline internal/extsvc/gitlab/projects.go:32]: This was added to maintain consistency with what we're capturing from GitHub repos. We're not making use of it in this PR.

**Comment by rmmh** [comment]
Rather than "stars", we should name this column "priority" and make it a float64, since ranking is not directly star counts.

**Comment by tsenart** [comment]
> Rather than "stars", we should name this column "priority" and make it a float64, since ranking is not directly star counts.

What are the plans for how this `priority` will be computed in the future? What other inputs will go into it? Additionally, how do user's search contexts affect these global priorities?

**Comment by tsenart** [comment]
@rmmh: If global repo rank computation isn't simple and requires more inputs than available from a single repo's fields, then I don't think we should store this rank/priority within the `repo` table, nor make repo syncing the mechanism through which we'd compute those values.

For making repo type search results and suggestions work efficiently, I think we need these priorities well modelled and indexed in Postgres, but that doesn't have to be in the repo table.

I can image some background job that subscribes to changes in the input data that is relevant for the per repo global rank computation, computes them and writes to a separate table like `repo_global_ranks`, which we can efficiently join against in `RepoStore.List`.

The same architecture could be used for per-user ranking potentially.

**Review by ryanslade — APPROVED** [review]
LGTM

I'm a little nervous about this starting a trend to add more columns to the repo table but your motivation for doing so that you describe makes senses and I guess is the better of two evils :)

**Review by rmmh — APPROVED** [review]

**Review by keegancsmith — APPROVED** [review]
love reading PRs which add a column, it feels like you are reading a mantra. In this case it is "stars" "stars" "stars" which feels quite positive :)

File: internal/extsvc/gitlab/projects.go (hunk 1 above)
Comment (efritz) [reply internal/extsvc/gitlab/projects.go:32]: Is this something that should be indexed now even if it's not used so we don't forget one later?

Comment (tsenart) [reply internal/extsvc/gitlab/projects.go:32]: @efritz: We can always add another column `forks` and an index to it just like we're adding `stars` in this PR, if needed?

Comment (efritz) [reply internal/extsvc/gitlab/projects.go:32]: Oops, sorry, I thought the column was there already but wasn't indexed. I misread.

Comment (tsenart) [reply internal/extsvc/gitlab/projects.go:32]: No problem! Thanks for reading :)

**Pull request merged**